otwcode/otwarchive
# AO3-5613 Renames Warning class to ArchiveWarning

# Pull Request Checklist


## Issue

https://otwarchive.atlassian.net/browse/AO3-5613 (please fill in issue number and remove this comment)

## Purpose

This pull requests renames Warning class to ArchiveWarning to prep for upcoming Ruby upgrades. In addition, this pull requests repairs broken tests as a result of the name change. Given there is quite a bit of meta programming around tags and warnings and the fact that "ArchiveWarning" introduced two words instead of one, required some regex updates and other updates to accommodate the new change. 

It should be noted that every was change was made fo "Warning" -> "ArchiveWarming" and warning  -> archive_warning, etc. In order to avoid confusion with Ruby's warning.

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended? (If you have access, please copy this into the JIRA ticket for them!)

I've verified that all tests pass on TravisCI.

## References

Are there any other relevant issues / PRs / mailing lists discussions related to this?

Not to my knowledge

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*

(If you have an account on JIRA, please include the same name in the "Full name" field on your JIRA profile,so we can find you and assign you the issues you're working on.)

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*

he/him

(If you do not fill this in, we will use your GitHub account name and will use "they" to avoid making assumptions about your gender.)
